### PR TITLE
Integration test tutorial

### PR DESCRIPTION
**Pull request by cmcantalupo**
Simple test that checks that the tutorials build and execute.

### CODE REVIEW

File: test_integration/test_tutorial_base.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        """Clean up any files that may have been created during the test if we
+        are not handling an exception and the GEOPM_KEEP_FILES
+        environment variable is unset.
+
+        """
+        if (sys.exc_info() == (None, None, None) and not
+            cls._keep_files):
+            tmp_dir = os.readlink(cls._tmp_link)
+            os.rmtree(tmp_dir)
+            os.unlink(cls._tmp_link)
```
Comment (bgeltz) [inline test_integration/test_tutorial_base.py]: The behavior of saving files when a test is failing is broken here.  The issue is described in #1054.

File: test_integration/test_tutorial_base.sh [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+#  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+#  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+#  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+#  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+#TUTORIAL_BASE
+#
+#  Setup script that installs geopm into a temporary location, and the
```
Comment (bgeltz) [inline test_integration/test_tutorial_base.sh]: *and then

File: test_integration/test_tutorial_base.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+import sys
+import unittest
+import os
+import glob
+import subprocess
+import matplotlib
+matplotlib.use('Agg')
+import matplotlib.pyplot as plt
+import pandas
```
Comment (bgeltz) [inline test_integration/test_tutorial_base.py]: Not used.

**Review by bgeltz — CHANGES_REQUESTED** [review]

File: test_integration/test_tutorial_base.py (hunk 1 above)
Comment (bgeltz) [reply test_integration/test_tutorial_base.py]: This seems to resolve the bug:
https://github.com/bgeltz/geopm/compare/7c2792c3c41d...c047f8a4a6a8

Comment (cmcantalupo) [reply test_integration/test_tutorial_base.py]: The code you got rid of is supposed to handle the case where an exception was thrown (sys.exc_info()).  Now that is kind of broken with python2 but we have a reset function to deal with that.

File: test_integration/test_template.py.in [hunk 4]
```diff
@@ -136,7 +136,7 @@ class TestIntegration_@test_name@(unittest.TestCase):
 
         """
 
-        if (sys.exc_info() == (None, None, None) and not
+
             cls._keep_files and not
```
Comment (bgeltz) [inline test_integration/test_template.py.in]: What happened to the if()?

**Review by bgeltz — CHANGES_REQUESTED** [review]

File: test_integration/test_template.py.in (hunk 4 above)
Comment (cmcantalupo) [reply test_integration/test_template.py.in]: Good catch.  I tried tested the template properly after fixing this.  Seems to work correctly.

**Review by cmcantalupo — COMMENTED** [review]
Runs though on mcfly.  This should now be good to go.

**Review by bgeltz — APPROVED** [review]

**Pull request merged**